File: C++/Relationship.cpp
#include <iostream>
#include <vector>
#include <string>

struct Player {
    std::string name;
    std::string address;
};

class Team {
    std::vector<Player*> players;
public:
    void set_player(Player* p) {
        players.push_back(p);  // aggregation
        // players.push_back(new Player());  // composition
    }
};

int main() {
    Player pradeep;
    Team team;
    team.set_player(&pradeep);
    return 0;
}

File: Python/Relationship.py
from typing import List


class Player:
    def __init__(self):
        self.name: str = ""
        self.address: str = ""


class Team:
    def __init__(self):
        self.players: List[Player] = []

    def set_player(self, p: "Player") -> None:
        self.players.append(p)  # aggregation
        # self.players.append(Player())  # composition


if __name__ == "__main__":
    pradeep = Player()
    team = Team()
    team.set_player(pradeep)
